src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -1344,6 +1498,35 @@ class spell_dru_t10_restoration_4p_bonus_dummy : public SpellScriptLoader
     }
 };
 
+// 77758 - Thrash
+class spell_dru_thrash : public SpellScript
+{
+    PrepareSpellScript(spell_dru_thrash);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_BEAR_FORM });
+    }
+
+    void HandleOnHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        if (Unit* hitUnit = GetHitUnit())
+        {
+            Unit* caster = GetCaster();
+
+            caster->CastSpell(hitUnit, SPELL_DRUID_THRASH_BEAR_AURA, TRIGGERED_FULL_MASK);
+
+            if (GetHitDamage() > 0 && caster->HasAura(SPELL_DRUID_BLOOD_FRENZY_AURA))
```

Comment:
Blood frenzy should trigger only from periodic damage from thrash (turn into proc, set periodic flag only)